spec/models/regulator_spec.rb

```diff
@@ -92,4 +106,53 @@
       end
     end
   end
+
+  describe "#batch_inspect!" do
+    context "with a few records" do
+      before do
+        ActiveRecord::Base.transaction do
+          10.times { create(:reading, :day_time, {temp: 75, outdoor_temp: 50 }) }
+        end
+      end
+
+      it "updates violation status in batches", :vcr do
+        expect(Reading.where(violation: true)).to have(0).items
+
+        Reading.order(created_at: :asc).first.update(temp: 65)
+        Reading.order(created_at: :asc).last.update(temp: 65)
+
+        Regulator.new(Reading.all).batch_inspect!(silent: true)
+        expect(Reading.where(violation: true)).to have(2).items
+      end
+    end
+
+    context "with many records" do
```

Comment:
what's meant to be tested here, that's not tested above in the `few records` context?